### PR TITLE
Add CAN client

### PR DESCRIPTION
**Pull request by Acuadros95**
TODO:
- [ ] Define CAN identifier for micro-ROS and filter other messages.
- [x] Add to integration XRCE tests.
- [x] Add example.

### CODE REVIEW

File: examples/PublishHelloWorldCAN/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    cmake_policy(SET CMP0048 NEW)
+endif()
+
+project(PublishHelloWorldClientCAN C)
+
+if(NOT UCLIENT_BUILD_EXAMPLES)
+    find_package(microxrcedds_client REQUIRED)
+endif()
+
+if(NOT UCLIENT_PROFILE_UDP)
```
Comment (pablogs9) [inline examples/PublishHelloWorldCAN/CMakeLists.txt:26]: This label should be UCLIENT_PROFILE_CAN

File: examples/PublishHelloWorldCAN/README.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+To run the example, a virtual CAN FD bus can be created with the following commands:
+```
+ip link add dev vcan0 type vcan
+ip link set vcan0 mtu 72
+ip link set dev vcan0 up
+```
+
+## Usage
+1. Run an agent in can mode, for example, *2021*: `MicroXRCEAgent canfd --dev vcan0 --id 0x00000001`.
```
Comment (pablogs9) [inline examples/PublishHelloWorldCAN/README.md:16]: What is the `for example 2021`?

File: src/c/profile/transport/can/can_transport_posix.c [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+                platform->can_id = can_id | CAN_EFF_FLAG;
+
+                // Add CAN filter for micro-ROS data
+                struct can_filter rfilter;
+                rfilter.can_id = platform->can_id;
+                rfilter.can_mask = CAN_EFF_MASK;
+                setsockopt(platform->poll_fd.fd, SOL_CAN_RAW, CAN_RAW_FILTER, &rfilter, sizeof(rfilter));
+                rv = true;
+            }
+
```
Comment (pablogs9) [inline src/c/profile/transport/can/can_transport_posix.c:50]: Extra spaces here

File: src/c/profile/transport/can/can_transport_posix.c [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+}
+
+size_t uxr_write_can_data_platform(
+        uxrCANPlatform* platform,
+        const uint8_t* buf,
+        size_t len,
+        uint8_t* errcode)
+{
+    struct canfd_frame frame = {
+        .can_id = platform->can_id, .len = (uint8_t)len
```
Comment (pablogs9) [inline src/c/profile/transport/can/can_transport_posix.c:70]: Where is guaranteed that this `len` cast is not going to overflow?

**Review by pablogs9 — CHANGES_REQUESTED** [review]

File: src/c/profile/transport/can/can_transport_posix.c (hunk 4 above)
Comment (Acuadros95) [reply src/c/profile/transport/can/can_transport_posix.c:70]: I assumed that `len` will never be bigger than the transport MTU, which fits inside an `uint8_t`:
https://github.com/eProsima/Micro-XRCE-DDS-Client/blob/0f4019b01cc1851c92d7a9e4e19ec2ed6c7d4729/include/uxr/client/profile/transport/can/can_transport.h#L32

File: examples/PublishHelloWorldCAN/README.md (hunk 2 above)
Comment (Acuadros95) [reply examples/PublishHelloWorldCAN/README.md:16]: All the examples add a reference year for the command ([example](https://github.com/eProsima/Micro-XRCE-DDS-Client/blob/0f4019b01cc1851c92d7a9e4e19ec2ed6c7d4729/examples/MultiSessionHelloWorld/README.md)).

Maybe we can update the commands if needed and remove this year references.

File: examples/PublishHelloWorldCAN/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    message(WARNING "Can not compile example: The UCLIENT_PROFILE_UDP must be enabled.")
```
Comment (Acuadros95) [inline examples/PublishHelloWorldCAN/CMakeLists.txt:27]:
```suggestion
if(NOT UCLIENT_PROFILE_CAN)
    message(WARNING "Can not compile example: The UCLIENT_PROFILE_CAN must be enabled.")
```

File: examples/PublishHelloWorldCAN/CMakeLists.txt (hunk 1 above)
Comment (Acuadros95) [reply examples/PublishHelloWorldCAN/CMakeLists.txt]: Fixed 85039e28d79cfd30d4ad32e89451a0c584f145b6

File: examples/PublishHelloWorldCAN/README.md (hunk 2 above)
Comment (Acuadros95) [reply examples/PublishHelloWorldCAN/README.md:16]:
```suggestion
1. Run an agent in can mode, for example: `MicroXRCEAgent canfd --dev vcan0 --id 0x00000001`.
```

File: src/c/profile/transport/can/can_transport_posix.c [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+                // Add CAN filter for micro-ROS data
+                struct can_filter rfilter;
+                rfilter.can_id = platform->can_id;
+                rfilter.can_mask = CAN_EFF_MASK;
+                setsockopt(platform->poll_fd.fd, SOL_CAN_RAW, CAN_RAW_FILTER, &rfilter, sizeof(rfilter));
+                rv = true;
+            }
+
+
+        }
```
Comment (Acuadros95) [inline src/c/profile/transport/can/can_transport_posix.c:52]:
```suggestion
                rv = true;
            }
        }
```

File: src/c/profile/transport/can/can_transport_posix.c (hunk 3 above)
Comment (Acuadros95) [reply src/c/profile/transport/can/can_transport_posix.c]: 6290208464aed1f6af256b570db1919036488a0a

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Client%20Manual%20Linux/475/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Client%20Manual%20Windows/489/)

**Review by pablogs9 — APPROVED** [review]

**Pull request merged**